### PR TITLE
[BugFix] BE maybe crash when we delete all records in PrimaryKey table with persistent index in branch-2.5+

### LINKED ISSUE #18616: [Bug] BE maybe crash when we delete all records in PrimaryKey table with persistent index in branch-2.5+
### Steps to reproduce the behavior (Required)
1.  Create PrimaryKey table with persistent index
 ```
CREATE TABLE `batch_flush_test` (
  `col_1` bigint(20) NOT NULL COMMENT "",
  `col_2` bigint(20) NOT NULL COMMENT "",
  `col_3` bigint(20) NOT NULL COMMENT "",
  `col_4` bigint(20) NOT NULL COMMENT "",
  `col_5` bigint(20) NOT NULL COMMENT "",
  `col_6` varchar(150) NULL COMMENT "",
  `col_7` varchar(150) NULL COMMENT ""
) ENGINE=OLAP 
PRIMARY KEY(`col_1`, `col_2`, `col_3`, `col_4`, `col_5`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`col_1`, `col_2`) BUCKETS 1 
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "V2",
"enable_persistent_index" = "true",
"compression" = "LZ4"
);
```

3. Importing large amounts of data(100M rows)
4. Delete all data in step 2

### Expected behavior (Required)
run success
### Real behavior (Required)
image

### StarRocks version (Required)
 - branch 2.5+

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/18616

## Problem Summary(Required) ：

This bug is introduced by the pr(https://github.com/StarRocks/starrocks/pull/15729) which enable flush persistent index in advance.

If we delete a record and we will keep the record which value is NullIndexValue. In following `_merge_compaction`, we will merge them and skip the old record if new record value is NullIndexValue. If we import a lot of data at once, we will select some tmp l1 file do merge compaction in advance and the delete operation in these files will be skip, but there are maybe some record in older l1 files which should be delete. So the delete operation will lost in merge compaction advance.  In the following merge compaction, we will create an empty shard if all records in this shard are delete, but we maybe read some expired record which should be delete which will cause BE crash.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by decster — DISMISSED** [review]

File: be/src/storage/persistent_index.cpp [hunk 1]
```diff
@@ -3261,7 +3372,7 @@ Status PersistentIndex::_merge_compaction() {
     const std::string idx_file_path =
             strings::Substitute("$0/index.l1.$1.$2", _path, _version.major(), _version.minor());
     RETURN_IF_ERROR(writer->init(idx_file_path, _version, true));
-    RETURN_IF_ERROR(_merge_compaction_internal(writer.get(), 0, _l1_vec.size(), _usage, _size));
+    RETURN_IF_ERROR(_merge_compaction_internal(writer.get(), 0, _l1_vec.size(), _usage, _size, false));
     // _usage should be equal to total_kv_size. But they may be differen because of compatibility problemw when we upgrade
```
[the remark this answers is not in the document]
Comment (sevev) [reply be/src/storage/persistent_index.cpp]: > spelling error. `problem`

done.

**Comment by sevev** [comment]
run starrocks_admit_test

**Review by caneGuy — APPROVED** [review]

**Comment by sevev** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**